thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/CompositePipelineConfigTranslator.java

```diff
@@ -309,30 +309,30 @@ private void fillInDetectorWrapperProperties(Map<String, Object> properties, Map
           properties.put(PROP_WINDOW_SIZE, 6);
           properties.put(PROP_WINDOW_UNIT, TimeUnit.HOURS);
       }
-      // override from yaml
-      if (yamlConfig.containsKey(PROP_WINDOW_SIZE)) {
-        properties.put(PROP_MOVING_WINDOW_DETECTION, true);
-        properties.put(PROP_WINDOW_SIZE, MapUtils.getString(yamlConfig, PROP_WINDOW_SIZE));
-      }
-      if (yamlConfig.containsKey(PROP_WINDOW_UNIT)) {
-        properties.put(PROP_MOVING_WINDOW_DETECTION, true);
-        properties.put(PROP_WINDOW_UNIT, MapUtils.getString(yamlConfig, PROP_WINDOW_UNIT));
-      }
-      if (yamlConfig.containsKey(PROP_WINDOW_DELAY)) {
-        properties.put(PROP_WINDOW_DELAY, MapUtils.getString(yamlConfig, PROP_WINDOW_DELAY));
-      }
-      if (yamlConfig.containsKey(PROP_WINDOW_DELAY_UNIT)) {
-        properties.put(PROP_WINDOW_DELAY_UNIT, MapUtils.getString(yamlConfig, PROP_WINDOW_DELAY_UNIT));
-      }
-      if (yamlConfig.containsKey(PROP_TIMEZONE)){
-        properties.put(PROP_TIMEZONE, MapUtils.getString(yamlConfig, PROP_TIMEZONE));
-      }
-      if (yamlConfig.containsKey(PROP_BUCKET_PERIOD)){
-        properties.put(PROP_BUCKET_PERIOD, MapUtils.getString(yamlConfig, PROP_BUCKET_PERIOD));
-      }
-      if (yamlConfig.containsKey(PROP_CACHE_PERIOD_LOOKBACK)) {
-        properties.put(PROP_CACHE_PERIOD_LOOKBACK, MapUtils.getString(yamlConfig, PROP_CACHE_PERIOD_LOOKBACK));
-      }
+    }
+    // override from yaml
+    if (yamlConfig.containsKey(PROP_WINDOW_SIZE)) {
+      properties.put(PROP_MOVING_WINDOW_DETECTION, true);
+      properties.put(PROP_WINDOW_SIZE, MapUtils.getString(yamlConfig, PROP_WINDOW_SIZE));
+    }
+    if (yamlConfig.containsKey(PROP_WINDOW_UNIT)) {
+      properties.put(PROP_MOVING_WINDOW_DETECTION, true);
+      properties.put(PROP_WINDOW_UNIT, MapUtils.getString(yamlConfig, PROP_WINDOW_UNIT));
+    }
+    if (yamlConfig.containsKey(PROP_WINDOW_DELAY)) {
```

Comment:
Currently there is a mix of three time units in YAML now and is confusing.
1. Milliseconds. E.g, max duration time.
2. Unit and value. E.g, window_deplay and window_delay_unit.
3. Period. E.g, bucket period.
Do we have plan to unify them?

Reply:
Agree. It's confusing now. The correct approach is to use Period. However, with the legacy notations already stored in config files, it's hard to eliminate them. We can come up with a way to use Period only in the future.